mne/datasets/utils.py

```diff
@@ -98,12 +98,12 @@ def _get_path(path, key, name):
     # 3. get_config('MNE_DATA')
     path = get_config(key or 'MNE_DATA', get_config('MNE_DATA'))
     if path is not None:
-        if not op.exists(path):
+        if not op.exists(op.expanduser(path)):
             msg = (f"Download location {path} as specified by MNE_DATA does "
                    f"not exist. Either create this directory manually and try "
                    f"again, or set MNE_DATA to an existing directory.")
             raise FileNotFoundError(msg)
-        return Path(path)
+        return Path(path).expanduser()
```

Comment:
(the aforementioned "corresponding change")

```suggestion
        return path
```